src/CrvInfoHelper.cc

```diff
@@ -0,0 +1,263 @@
+#include "TrkAna/inc/CrvInfoHelper.hh"
+
+#include "Offline/CosmicRayShieldGeom/inc/CosmicRayShield.hh"
+#include "Offline/CRVConditions/inc/CRVDigitizationPeriod.hh"
+#include "Offline/GeometryService/inc/DetectorSystem.hh"
+#include "Offline/GeometryService/inc/GeomHandle.hh"
+#include "Offline/GeometryService/inc/GeometryService.hh"
+#include "Offline/MCDataProducts/inc/CrvCoincidenceClusterMC.hh"
+#include "Offline/MCDataProducts/inc/CrvDigiMC.hh"
+#include "Offline/MCDataProducts/inc/CrvStep.hh"
+#include "Offline/MCDataProducts/inc/MCTrajectory.hh"
+#include "Offline/RecoDataProducts/inc/CrvCoincidenceCluster.hh"
+#include "Offline/RecoDataProducts/inc/CrvDigi.hh"
+#include "Offline/RecoDataProducts/inc/CrvRecoPulse.hh"
+#include "Offline/DataProducts/inc/PDGCode.hh"
+#include "art/Framework/Principal/Handle.h"
+#include "Offline/CRVResponse/inc/CrvMCHelper.hh"
+#include "Offline/CRVReco/inc/CrvHelper.hh"
+#include "Offline/GeometryService/inc/DetectorSystem.hh"
+
+namespace mu2e
+{
+  void CrvInfoHelper::FillCrvHitInfoCollections(
+      art::Handle<CrvCoincidenceClusterCollection> const& crvCoincidences,
+      art::Handle<CrvCoincidenceClusterMCCollection> const& crvCoincidencesMC,
+      art::Handle<CrvRecoPulseCollection> const& crvRecoPulses,
+      art::Handle<CrvStepCollection> const& crvSteps,
+      art::Handle<MCTrajectoryCollection> const& mcTrajectories,
+      CrvHitInfoRecoCollection &recoInfo, CrvHitInfoMCCollection &MCInfo,
+      CrvSummaryReco &recoSummary, CrvSummaryMC &MCSummary,
+      CrvPlaneInfoMCCollection &MCInfoPlane, double crvPlaneY) {
+    GeomHandle<CosmicRayShield> CRS;
+    GeomHandle<DetectorSystem> tdet;
+    if(!crvCoincidences.isValid()) return;
+    size_t nClusters=crvCoincidences->size();
+    for(size_t i=0; i<nClusters; i++)
+    {
+      const CrvCoincidenceCluster &cluster = crvCoincidences->at(i);
+
+      //fill the Reco collection
+      recoInfo.emplace_back(
+          cluster.GetCrvSectorType(),
+          tdet->toDetector(cluster.GetAvgHitPos()),
```

Comment:
a cluster.GetAvgHitTime() should be added to be consistent.

Reply:
will do